tests/CustomMatchers/CustomMatchersForTaskSerializer.ts

```diff
@@ -83,6 +86,10 @@ function summarizeTaskDetails(t: TaskDetails | null): SummarizedTaskDetails | nu
         scheduledDate: t.scheduledDate?.format(TaskRegularExpressions.dateFormat) ?? null,
         dueDate: t.dueDate?.format(TaskRegularExpressions.dateFormat) ?? null,
         doneDate: t.doneDate?.format(TaskRegularExpressions.dateFormat) ?? null,
+        reminderDate:
+            t.reminderDate?.format(
+                isDateTime(t.reminderDate) ? TaskRegularExpressions.dateTimeFormat : TaskRegularExpressions.dateFormat,
```

Comment:
This pattern of `isDateTime() ? ... : ...` is going to grow and grow....

we could save our time by introducing a helper function that does the check and the formatting...